pkg/cmd/cli/cmd/factory.go

```diff
@@ -86,19 +81,32 @@ func NewFactory(clientConfig clientcmd.ClientConfig) *Factory {
 
 // Clients returns an OpenShift and Kubernetes client.
 func (f *Factory) Clients(cmd *cobra.Command) (*client.Client, *kclient.Client, error) {
-	os, err := f.OpenShiftClientConfig.ClientConfig()
+	cfg, err := f.OpenShiftClientConfig.ClientConfig()
 	if err != nil {
 		return nil, nil, err
 	}
-	oc, err := client.New(os)
+
+	transport, err := kclient.TransportFor(cfg)
```

Comment:
You want to create a transport and pass it in to client.New on the config.  Your transport should wrap http.DefaultTransport.

----- Original Message -----
> >  	if err != nil {
> >  		return nil, nil, err
> >  	}
> > -	oc, err := client.New(os)
> > +
> > +	transport, err := kclient.TransportFor(cfg)
> 
> @smarterclayton I need to intercept every API request in order to check for
> an eventual *401 Unauthorized* and then be able to print a proper,
> user-friendly, message (*you need to log in bla bla*). Any feedback about
> this approach? Would you suggest something different than taking the clients
> and injecting it here in the Factory? Looks a little hacky, nothing protects
> us against people doing `client.New` / `kclient.New` directly.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/992/files#r24695535